x-pack/plugins/case/common/api/cases/comment.ts

```diff
@@ -18,22 +33,42 @@ export const CommentAttributesBasicRt = rt.type({
   updated_by: rt.union([UserRT, rt.null]),
 });
 
+export enum CommentType {
+  user = 'user',
+  alert = 'alert',
+  generatedAlert = 'generated_alert',
+}
+
 export const ContextTypeUserRt = rt.type({
   comment: rt.string,
-  type: rt.literal('user'),
+  type: rt.literal(CommentType.user),
 });
 
 export const ContextTypeAlertRt = rt.type({
-  type: rt.literal('alert'),
-  alertId: rt.string,
+  type: rt.union([rt.literal(CommentType.generatedAlert), rt.literal(CommentType.alert)]),
+  alertId: rt.union([rt.array(rt.string), rt.string]),
+  index: rt.string,
+});
+
+const AlertIDRt = rt.type({
+  _id: rt.string,
+});
+
+export const ContextTypeGeneratedAlertRt = rt.type({
+  type: rt.literal(CommentType.generatedAlert),
+  alerts: rt.union([rt.array(AlertIDRt), AlertIDRt]),
   index: rt.string,
 });
 
 const AttributesTypeUserRt = rt.intersection([ContextTypeUserRt, CommentAttributesBasicRt]);
 const AttributesTypeAlertsRt = rt.intersection([ContextTypeAlertRt, CommentAttributesBasicRt]);
 const CommentAttributesRt = rt.union([AttributesTypeUserRt, AttributesTypeAlertsRt]);
 
-const ContextBasicRt = rt.union([ContextTypeUserRt, ContextTypeAlertRt]);
+const ContextBasicRt = rt.union([
+  ContextTypeUserRt,
+  ContextTypeAlertRt,
+  ContextTypeGeneratedAlertRt,
```

Comment:
Correct me if I am wrong but to my understanding `ContextTypeGeneratedAlertRt` is only used by alerts generated from a rule and they are saved to ES as `ContextTypeAlertRt`. Why we allow users to edit alerts of `ContextTypeGenerated` if that's the case?

Reply:
Good catch, I changed the patch so it can only accept user and alert and not generated alerts. I'm leaving generated alerts here so we don't need a separate type for using the post comment api and the connector api. So technically a user _could_ use the post API to add a "generated" alert to a collection. I think we can just leave it out of the documentation for now 🤷 .